libcudacxx/include/cuda/std/detail/libcxx/include/__config

```diff
@@ -47,17 +47,31 @@
 #  define _LIBCUDACXX_COMPILER_CLANG_CUDA
 #endif
 
-// Some convenience macros to filter nvcc versions
-#if defined(_LIBCUDACXX_COMPILER_NVCC) && _LIBCUDACXX_CUDACC_VER < 1102000
-#define _LIBCUDACXX_COMPILER_NVCC_BELOW_11_2
-#endif // defined(_LIBCUDACXX_COMPILER_NVCC) && _LIBCUDACXX_CUDACC_VER < 1102000
-#if defined(_LIBCUDACXX_COMPILER_NVCC) && _LIBCUDACXX_CUDACC_VER < 1103000
-#define _LIBCUDACXX_COMPILER_NVCC_BELOW_11_3
-#endif // defined(_LIBCUDACXX_COMPILER_NVCC) && _LIBCUDACXX_CUDACC_VER < 1103000
-
-#if defined(_LIBCUDACXX_COMPILER_NVCC) && _LIBCUDACXX_CUDACC_VER < 1108000
-#define _LIBCUDACXX_COMPILER_NVCC_BELOW_11_8
-#endif // defined(_LIBCUDACXX_COMPILER_NVCC) && _LIBCUDACXX_CUDACC_VER < 1108000
+#ifdef __CUDACC__
+#  define _LIBCUDACXX_CUDACC
+#  define _LIBCUDACXX_CUDACC_VER_MAJOR __CUDACC_VER_MAJOR__
+#  define _LIBCUDACXX_CUDACC_VER_MINOR __CUDACC_VER_MINOR__
+#  define _LIBCUDACXX_CUDACC_VER_BUILD __CUDACC_VER_BUILD__
+#  define _LIBCUDACXX_CUDACC_VER                                                  \
+      _LIBCUDACXX_CUDACC_VER_MAJOR * 100000 + _LIBCUDACXX_CUDACC_VER_MINOR * 1000 + \
+      _LIBCUDACXX_CUDACC_VER_BUILD
+
+// TODO: Determine if this is necessary, I don't know why we automatically include this in <config>
+#  if defined(__clang__) || defined(__FLT16_MANT_DIG__)
+#    include <cuda_fp16.h>
```

Comment:
I'm guessing it is for nvrtc.

Reply:
`__clang__` wouldn't be defined for NVRTC?

Reply:
Maybe the FLT16 flag would be... That seems like a weird detection mechanism though.